inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -468,7 +468,7 @@ void MKLDNNPlugin::MKLDNNInferRequest::SetBlob(const std::string& name, const In
                                << blobDesc.getPrecision() << ", if CNNNetwork output blob precision is: " << foundOutput->getPrecision();
         }
         size_t outputSize = foundOutput->getTensorDesc().getLayout() != InferenceEngine::Layout::SCALAR
-            ? InferenceEngine::details::product(foundOutput->getDims())
+            ? InferenceEngine::details::product(foundOutput->getTensorDesc().getDims())
```

Comment:
why is it needed?

Reply:
hi @ilya-lavrenov , in auto/multi plugin, we will need to call SetBlob/GetBlob for actual device infer request for both input/output buffers:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto/infer_request.cpp#L74

Reply:
I suppose `foundOutput->getDims()` is the same `foundOutput->getTensorDesc().getDims()`, is not it?

Reply:
@ilya-lavrenov no, using foundOutput->getDims() will throw exception on below, this API does not support dynamic shape:

Expected: req.infer() doesn't throw an exception.
  Actual: it throws.

**/home/bell/work/mount/openvino/src/inference/src/ie_data.cpp:202 Cannot return dims for Data with dynamic shapes!**

Reply:
@maxnick why old API is used while dynamism can live internally?

Reply:
@ilya-lavrenov , to answer your question, yes we have the old API that may be used in dynamic context, and the reason is that when we were enabling dynamic shapes support in the plugin, the new API was not ready. Obviously, we have to refactor the API implementation using the new paradigm to avoid such problems in the future.
Regarding the change itself, it seems that it is better to leave the `outputSize` computation as it was before, but move it under `if(!isDynamic)` guard check on L475 since the `outputSize` only makes sense under this check.

Reply:
fixed, thanks for review @maxnick